Use new ordered set to allow for controlling strictMode (#984)
flame-engine/flame: packages/flame/CHANGELOG.md

 - Method `PositionComponent.toRect()` now works for flipped/rotated components.
 - Make the root bundle exposed via `Flame.bundle` actually configurable
 - Take in an optional `Camera` as a parameter to `FlameGame`
 - Make super.onLoad mandatory to avoid memory leaks

>  - QueryableOrderedSet strictMode is configurable so you don't need to call `register` before `query`

Reviewer: ```suggestion
 - `QueryableOrderedSet`'s `strictMode` is configurable so it is no longer necessary to call `register` before `query`
```
